### DISCUSSION TITLE
Setting the default project

### DISCUSSION THREAD

**Discussion by abcxcba** [goharbor/harbor#20476] [Ideas]
I've seen the "default project name" question raised many times:

Basically expect a command similar to " **docker pull redis:6.0.8**" to execute.

It goes without saying that the docker hub can pull and push without a project name.
docker pull registry.k8s.io/kube-apiserver:v1.30.0 will also download the image.

All the replies to this question here at harbor say "_you must add the project name_ ".

I think the **code logic**:

**An anonymous user request:**
When parsing the url, insert "library" or the system set "defalut project" name if there is no project name.
eg: reg.example.com/redis:latest ---> reg.example.com/library/redis:latest

**Login user request:**
When parsing the url, the user defaults to the "defalut project" name if there is no project name.
eg: reg.example.com/redis:latest ---> reg.example.com/user_default_project/redis:latest

**Determine if there is a project name**: If there is only a "/" after the domain, it is assumed that there is no project name.

Isn't it very difficult to implement in code?
Wouldn't it be easier to use this feature?

**Comment by spadeq** [comment]
This feature is extremely important. I have to deploy another standard Docker Registry to serve such "default" images.
